### PR TITLE
feat: Ignore files from the validation

### LINKED ISSUE #887: Add files / folders with a leading underscore as an excluded type in trestle author * validate
## Issue description / feature objectives
Currently we exclude, by default, hidden files as defined by the OS (e.g. a leading period on POSIX systems or using windows hidden file attributes).

When considering a web application, hidden files are typically completely ignored, however, there is a concept of 'supporting files', which are by convention typically preceded by an underscore. 

The proposal here is that files which start with a leading underscore are *NOT* indexed. The open question is whether this should be a flag or enabled universally. This would obviously include any file under a directory which begins with an underscore.

NOTE: I believe this should be a flag to maintain backwards compatibility.

### PR DESCRIPTION
**Pull request by enikonovad**
Signed-off-by: Ekaterina Nikonova <enikonovad@gmail.com>

## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary
Ignore files and folders that match given regex via new flag `-ig` `--ignore` from the validation of `docs|headers|folders`. 

Example:
`trestle author docs validate -tn my_task_name -ig ^_.*`

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar

Closes #887

### CODE REVIEW

File: docs/trestle_author.md [hunk 1]
```diff
@@ -209,6 +217,10 @@ Validate the headers in markdown and metadata in drawio files.
 
 Turns off the validation of the structure of the document and only validates the yaml header structure and drawio files.
 
+#### Ignore files or folders from validation (`-ig`, `--ignore`)
+
+Provide regular expression after this flag to ignore folders and files which names match this regular expression during **validation only**.
+
```
Comment (butler54) [inline docs/trestle_author.md:223]: Should we deprecate exclude in this case?

Comment (butler54) [reply docs/trestle_author.md:223]: (mark as deprecated).

**Review by butler54 — APPROVED** [review]
LGTM

File: docs/trestle_author.md (hunk 1 above)
Comment (enikonovad) [reply docs/trestle_author.md:223]: Provided a mechanism to deprecate arguments for the future purposes

Comment (enikonovad) [reply docs/trestle_author.md:223]: Added a warning

**Review by butler54 — APPROVED** [review]
LGTM

**Pull request merged**